### ISSUE TITLE
a11y: logged-in user avatar is an unlabelled button (WCAG level A)

### ISSUE THREAD

**Issue by eatyourgreens** [OxfordRSE/gutenberg#402]
When you're logged in, the user account avatar is an unlabelled button. The accessible label is on the parent list item, but should be on the button itself (or the child image.)

<img width="708" height="349" alt="Screenshot of the logged-in user avatar, with the accessibility inspector open to show that it's an unlabelled button." src="https://github.com/user-attachments/assets/78632fa3-d352-4b30-a954-93e62b47d26f" />

Related WCAG SC: [1.1.1 Non-text content](https://www.w3.org/WAI/WCAG22/Understanding/non-text-content.html) (level A.)
